relayer/msgs.go

```diff
@@ -0,0 +1,158 @@
+package relayer
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	transfertypes "github.com/cosmos/cosmos-sdk/x/ibc/applications/transfer/types"
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/core/02-client/types"
+	conntypes "github.com/cosmos/cosmos-sdk/x/ibc/core/03-connection/types"
+	chantypes "github.com/cosmos/cosmos-sdk/x/ibc/core/04-channel/types"
+	ibcexported "github.com/cosmos/cosmos-sdk/x/ibc/core/exported"
+)
+
+// NOTE: we explicitly call 'MustGetAddress' before 'NewMsg...'
+// to ensure the correct config file is being used to generate
+// the account prefix. 'NewMsg...' functions take an AccAddress
+// rather than a string. The 'address.String()' function uses
+// the currently set config file. Querying a counterparty would
+// swap the config file. 'MustGetAddress' sets the config file
+// correctly. Do not change this ordering until the SDK config
+// file handling has been refactored.
+// https://github.com/cosmos/cosmos-sdk/issues/8332
+
+// UpdateClient creates an sdk.Msg to update the client on src with data pulled from dst
+func (c *Chain) UpdateClient(dstHeader ibcexported.Header) sdk.Msg {
+	if err := dstHeader.ValidateBasic(); err != nil {
+		panic(err)
```

Comment:
can we return an error instead?

Reply:
this was the existing structure. I'd rather not change too much in one pr